shardingsphere-features/shardingsphere-encrypt/shardingsphere-encrypt-core/src/main/java/org/apache/shardingsphere/encrypt/rewrite/token/generator/EncryptAlterTableTokenGenerator.java

```diff
@@ -189,42 +244,89 @@ private EncryptAlterTableToken getPositionColumnToken(final ColumnPositionSegmen
         Collection<SQLToken> result = new LinkedList<>();
         for (ChangeColumnDefinitionSegment each : changeColumnDefinitions) {
-            result.addAll(getPreviousColumnTokens(tableName, each));
-            result.addAll(getColumnTokens(tableName, each));
+            result.addAll(getChangeColumnTokensEach(tableName, each));
             each.getColumnPosition().flatMap(columnPositionSegment -> getColumnPositionToken(tableName, columnPositionSegment)).ifPresent(result::add);
-            if (null != each.getPreviousColumn()) {
-                String previousColumnName = each.getPreviousColumn().getIdentifier().getValue();
-                encryptRule.findAssistedQueryColumn(tableName, previousColumnName).map(optional ->
-                        new EncryptAlterTableToken(each.getStopIndex() + 1, each.getColumnDefinition().getColumnName().getStopIndex(),
-                                each.getColumnDefinition().getColumnName().getIdentifier().getValue() + "_assisted", ", CHANGE COLUMN " + optional)).ifPresent(result::add);
-                encryptRule.findPlainColumn(tableName, previousColumnName).map(optional ->
-                        new EncryptAlterTableToken(each.getStopIndex() + 1, each.getColumnDefinition().getColumnName().getStopIndex(),
-                                each.getColumnDefinition().getColumnName().getIdentifier().getValue() + "_plain", ", CHANGE COLUMN " + optional)).ifPresent(result::add);
-            }
         }
         return result;
     }
     
-    private Collection<? extends SQLToken> getColumnTokens(final String tableName, final ChangeColumnDefinitionSegment segment) {
-        if (null == segment.getPreviousColumn() || null == segment.getPreviousColumn().getIdentifier().getValue()
-                || !encryptRule.findEncryptor(tableName, segment.getPreviousColumn().getIdentifier().getValue()).isPresent()) {
+    private Collection<? extends SQLToken> getChangeColumnTokensEach(final String tableName, final ChangeColumnDefinitionSegment segment) {
+        checkAlterChangeIsSupport(tableName, segment);
+        if (!encryptRule.findEncryptor(tableName, segment.getPreviousColumn().getIdentifier().getValue()).isPresent() 
+                || !encryptRule.findEncryptor(tableName, segment.getColumnDefinition().getColumnName().getIdentifier().getValue()).isPresent()) {
             return Collections.emptyList();
         }
-        if (null == segment.getColumnDefinition() || null == segment.getColumnDefinition().getColumnName() || null == segment.getColumnDefinition().getColumnName().getIdentifier().getValue()) {
-            return Collections.emptyList();
+        Collection<SQLToken> result = new LinkedList<>();
+        result.addAll(getPreviousColumnTokens(tableName, segment));
+        result.addAll(getColumnTokens(tableName, segment));
+        return result;
+    }
+    
+    private void checkAlterChangeIsSupport(final String tableName, final ChangeColumnDefinitionSegment segment) {
```

Comment:
Do you think `isSameEncryptColumn` may be better?

Reply:
> Do you think `isSameEncryptColumn` may be better?

Yes, it's better.